src/components/ImportRecordsFlow/stepThree/__test__/assignFieldModal.spec.js

```diff
@@ -1,67 +1,227 @@
+            { label: 'Last_Name', value: 'Last_Name' },
+            { label: 'Email', value: 'Email' },
+        ]);
+    });
+    it('should set the right values in SelectedFieldsToAssign component when the param fieldsMap[databaseFieldToAssign] exists', () => {
+        const columns = ['First_Name', 'Last_Name', 'Email'];
+        const fieldsMap = { name: 'First_Name,Last_Name' };
+        const databaseFieldToAssign = 'name';
+        const component = mount(
+            <AssignFieldModal
+                isAssignFieldModalOpen
+                attributes={attributes}
+                columns={columns}
+                fieldsMap={fieldsMap}
+                databaseFieldToAssign={databaseFieldToAssign}
+            />,
+        );
+        expect(component.find(SelectedFieldsToAssign).prop('values')).toEqual([
+            'First_Name',
+            'Last_Name',
+        ]);
+    });
+    it('should set the right values in SelectedFieldsToAssign component when the param fieldsMap[databaseFieldToAssign] does not exists', () => {
+        const columns = ['First_Name', 'Last_Name', 'Email'];
+        const component = mount(
+            <AssignFieldModal isAssignFieldModalOpen attributes={attributes} columns={columns} />,
+        );
+        expect(component.find(SelectedFieldsToAssign).prop('values')).toEqual([]);
+    });
+    it('should fire onRequestClose when clicking assing button', () => {
+        const onRequestCloseFn = jest.fn();
+        const columns = ['First_Name', 'Last_Name', 'Email'];
+        const fieldsMap = { name: 'First_Name,Last_Name' };
+        const databaseFieldToAssign = 'name';
+        const component = mount(
+            <AssignFieldModal
+                isAssignFieldModalOpen
+                attributes={attributes}
+                onRequestClose={onRequestCloseFn}
+                fieldsMap={fieldsMap}
+                databaseFieldToAssign={databaseFieldToAssign}
+                columns={columns}
+            />,
+        );
+        component
+            .find(AssignFieldModalFooter)
+            .find(Button)
+            .at(1)
+            .simulate('click');
+        expect(onRequestCloseFn).toHaveBeenCalled();
+    });
+    it('should fire onAssignField with the right values when clicking assing button', () => {
+        const onAssignFieldFn = jest.fn();
+        const columns = ['First_Name', 'Last_Name', 'Email'];
+        const fieldsMap = { name: 'First_Name,Last_Name' };
+        const databaseFieldToAssign = 'name';
+        const component = mount(
+            <AssignFieldModal
+                isAssignFieldModalOpen
+                attributes={attributes}
+                onAssignField={onAssignFieldFn}
+                fieldsMap={fieldsMap}
+                databaseFieldToAssign={databaseFieldToAssign}
+                columns={columns}
+            />,
+        );
+        component
+            .find(AssignFieldModalFooter)
+            .find(Button)
+            .at(1)
```

Comment:
in this test and the above you use the same selector:
```.find(AssignFieldModalFooter).find(Button).at(1)```
to trigger onRequestClose and onAssignField
how is that possible?
I think you can use 'Button' and StyledCancelButton without the `at` method

Reply:
I had to do that way because StyledCancelButton is extended from a Button component, so the first one .at(0) is the Cancel Button.

```
```
<StyledCancelButton label="Cancel" onClick={onCancel} />
<Button
                label="Assign"
                variant="brand"
                onClick={onAssign}
                disabled={isAssignButtonDisabled}
/>
```